R/gh_token.R

```diff
@@ -142,3 +144,30 @@ obfuscate <- function(x, first = 4, last = 4) {
     substr(x, start = nchar(x) - last + 1, stop = nchar(x))
   )
 }
+
+get_connect_session <- function() {
+  if (!identical(Sys.getenv("RSTUDIO_PRODUCT"), "CONNECT")) {
+    return(NULL)
+  }
+  if (!isNamespaceLoaded("shiny")) {
+    return(NULL)
+  }
+  if (is_installed("connectcreds")) {
+    # Avoid taking a Suggests dependency on Shiny, which is otherwise
+    # irrelevant to gh.
+    f <- get("getDefaultReactiveDomain", envir = asNamespace("shiny"))
+    return(f())
+  }
+  cli::cli_inform(
```

Comment:
If this is running in Connect, where do you expect someone to see this message?

This also seems to assume that if I'm running on Connect, then I must want to use viewer credentials. What if I've put a PAT in an encrypted env var?

Reply:
It shows up in Connect's logs... which is not ideal, but it's all I can think of. Here is a screenshot of an earlier version of the message:

![Screenshot from 2024-12-05 14-49-41](https://github.com/user-attachments/assets/89fe877f-1e08-484b-9c70-188c637f6055)

And it's intentionally an `info` suggestion here, not a warning or error, because I absolutely expect that publishers will want to use PATs instead for some apps, too.